### ISSUE TITLE
`subspan` should accept `tuple` in addition to `pair`

### ISSUE THREAD

**Issue by dhollman** [kokkos/kokkos#2056]
By request at Oakland bootcamp

**Comment by dhollman** [comment]
Also, it should take a `std::array<Integral, 2>`.  Really, anything that works with `std::get<0>` and `std::get<1>`.

**Comment by mhoemmen** [comment]
Would this make the following work?
```c++
View<double**, ...> X (...);
// ...
auto X_sub = subspan (X, {0, numRows}, {0, numCols});
```

**Comment by dhollman** [comment]
@mhoemmen I don't think so. We can provide this, but we have to do it explicitly for all of the possibilities.  `initializer_list`s don't work well with deduced template parameters: <https://godbolt.org/z/v2JJMl>

**Comment by mhoemmen** [comment]
@dhollman Alas....

**Comment by dhollman** [comment]
It was also pointed out that `Kokkos::pair` is missing the necessary interface (intrusive `get` method) to work with `std::get`, so we should add that too.
